pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -3877,222 +3868,56 @@ func (s *PluginSuite) TestUpdateFederationRelationship() {
+					previousMinor.Minor--
+				}
+				return fmt.Sprintf(`
+					CREATE TABLE "migrations" ("id" integer primary key autoincrement, "version" integer,"code_version" varchar(255) );
+					INSERT INTO migrations("version", "code_version") VALUES (%d,%q);
+				`, schemaVersion, previousMinor)
+			}
 
-			attestedNode, err := s.ds.FetchAttestedNode(context.Background(), "spiffe://example.org/host")
-			s.Require().NoError(err)
+			prepareDB := func(migrationSupported bool) {
+				dump := migrationDumps[schemaVersion]
+				if migrationSupported {
+					require.NotEmpty(dump, "no migration dump set up for schema version")
+				} else {
+					require.Empty(dump, "migration dump exists for unsupported schema version")
+					dump = minimalDB()
+				}
+				dumpDB(t, dbPath, dump)
+				err := s.ds.Configure(fmt.Sprintf(`
+				database_type = "sqlite3"
+				connection_string = %q
+			`, dbURI))
+				if migrationSupported {
+					require.NoError(err)
+				} else {
+					require.EqualError(err, fmt.Sprintf("datastore-sql: migrating from schema version %d requires a previous SPIRE release; please follow the upgrade strategy at doc/upgrading.md", schemaVersion))
+				}
+			}
 
-			// Assert current serial numbers and expiration time remains the same
-			expectedTime, err := time.Parse(time.RFC3339, "2018-12-19T15:26:58-07:00")
-			s.Require().NoError(err)
-			s.Require().Equal(expectedTime.Unix(), attestedNode.CertNotAfter)
-			s.Require().Equal("111", attestedNode.CertSerialNumber)
-
-			// Assert the new fields are empty for pre-existing entries
-			s.Require().Empty(attestedNode.NewCertSerialNumber)
-			s.Require().Empty(attestedNode.NewCertNotAfter)
-		case 13:
-			s.Require().True(s.ds.db.Dialect().HasColumn("registered_entries", "revision_number"))
-		case 14:
-			db, err := openSQLite3(dbURI)
-			defer db.Close()
-			s.Require().NoError(err)
-			s.Require().True(db.Dialect().HasIndex("attested_node_entries", "idx_attested_node_entries_expires_at"))
-		case 15:
-			s.Require().True(s.ds.db.Dialect().HasColumn("registered_entries", "store_svid"))
-		case 16:
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "trust_domain"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "bundle_endpoint_url"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "bundle_endpoint_profile"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "endpoint_spiffe_id"))
-			s.Require().True(s.ds.db.Dialect().HasColumn("federated_trust_domains", "implicit"))
-			s.Require().True(s.ds.db.Dialect().HasIndex("federated_trust_domains", "uix_federated_trust_domains_trust_domain"))
-		default:
-			s.T().Fatalf("no migration test added for version %d", i)
-		}
+			switch schemaVersion {
+			// All of these schema versions were migrated by previous versions
+			// of SPIRE server and no longer have migration code.
+			case 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17:
```

Comment:
once we create new scripts that results in new schemas, when we move to next minor, we "must" add that version here and remove scripts?
if that is the case, may we add a comment to facilitate we remember it?

Reply:
Where would you like to see that comment? I mention in two spots about opportunistic removal of the migration. If we don't update this test, but remove the migrations, then the unit-tests will fail.